rclpy/test/test_qos_event.py

```diff
@@ -53,6 +54,9 @@ def setUp(self):
     def tearDown(self):
         self.node.destroy_node()
         rclpy.shutdown(context=self.context)
+        del self.context
+        del self.node
+        gc.collect()
```

Comment:
I see what you are doing here, but I'm not entirely sure why.  You are hand-destroying both objects, and then making sure they are collected so that the destructors run (and presumably release the underlying `rcl` resources).

But in the case of the node, shouldn't call `self.node.destroy_node()` above be enough?  If not, why not?  As for the context, maybe we should add a `destroy_context()` method as well?  Neither is very pythonic, but it at least gives users a deterministic way to destroy the objects, rather than forcing gc runs (which may have other consequences).

Reply:
> But in the case of the node, shouldn't calling self.node.destroy_node() above be enough? If not, why not?

I think it should be enough, but if I delete the `del self.node` line it stops working.
I will investigate a bit more to find the reason.

Reply:
See https://github.com/ros2/rclpy/pull/603/commits/ef3b7851ded2eea6f8f41711af2e385c78a7bd2d